slither/core/solidity_types/user_defined_type.py

```diff
@@ -8,10 +8,14 @@
     from slither.core.declarations.structure import Structure
     from slither.core.declarations.enum import Enum
     from slither.core.declarations.contract import Contract
+    from slither.core.declarations import EnumContract
+    from slither.core.declarations.structure_top_level import StructureTopLevel
 
 # pylint: disable=import-outside-toplevel
 class UserDefinedType(Type):
-    def __init__(self, t):
+    def __init__(
+        self, t: Union["EnumContract", "StructureTopLevel", "Contract", "StructureContract"]
```

Comment:
There's some confusion here about whether it's a Structure or Union[StructureTopLevel, StructureContract] between the type annotations and assertions